### PR TITLE
Remove devtools/archive

### PR DESCRIPTION
**Pull request by nd-02110114**
## Description

`devtools/archive` is not necessary, so I removed.
I made the note for the new release process in `notes/release.md`. 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by nd-02110114** [comment]
The CI error is not related, so I think it is ready to review.

File: devtools/archive/jenkins/results.table [hunk 1]
```diff
@@ -1,386 +0,0 @@
-Index splitting
```
Comment (rbharath) [inline devtools/archive/jenkins/results.table:1]: Would you be willing to make an issue in the https://github.com/deepchem/moleculenet and paste in this table for reference? @mufeili has been taking the lead on getting some new benchmark numbers up and these old numbers may be a useful sanity check for him as he's benchmarking

**Review by rbharath — APPROVED** [review]
Looks good to go! I've made one minor request to document the old benchmark numbers table in the https://github.com/deepchem/moleculenet repo but otherwise this is good to merge

File: devtools/archive/jenkins/results.table (hunk 1 above)
Comment (nd-02110114) [reply devtools/archive/jenkins/results.table:1]: OK 👍  I made deepchem/moleculenet#18.

**Pull request merged**